bin/eslint.js

```diff
@@ -57,6 +57,19 @@ process.once("uncaughtException", err => {
     process.exitCode = 2;
 });
 
+/**
+ * Handles the condition based Promise as returned by cli.execute()
+ * @param {string | void} arg argument to be passed in
+ * @returns {void}
+ */
+async function handleArgs(arg) {
+    const exitCode = cli.execute(process.argv, arg ? arg : null);
+
+    process.exitCode = exitCode instanceof Promise // eslint-disable-line
+        ? await exitCode
+        : exitCode;
+}
```

Comment:
All these test cases fail:-
```bash
1) bin/eslint.js
       reading from stdin
         gives a detailed error message if no config file is found in /:

      Expected an exit code of 2 but got 0.
      + expected - actual

      -0
      +2
      
      at /home/james-the-hacker/CodeSpace/GitHub-Projects/eslint/tests/bin/eslint.js:30:16
      at processTicksAndRejections (internal/process/task_queues.js:85:5)
      at async Promise.all (index 0)

  2) bin/eslint.js
       running on files
         has exit code 0 if a linting warning is reported:

      Expected an exit code of 0 but got 1.
      + expected - actual

      -1
      +0
      
      at /home/james-the-hacker/CodeSpace/GitHub-Projects/eslint/tests/bin/eslint.js:30:16
      at processTicksAndRejections (internal/process/task_queues.js:85:5)

  3) bin/eslint.js
       handling crashes
         prints the error message to stderr in the event of a crash:

      Expected an exit code of 2 but got 0.
      + expected - actual

      -0
      +2
      
      at /home/james-the-hacker/CodeSpace/GitHub-Projects/eslint/tests/bin/eslint.js:30:16
      at processTicksAndRejections (internal/process/task_queues.js:85:5)
      at async Promise.all (index 0)

  4) bin/eslint.js
       handling crashes
         prints the error message exactly once to stderr in the event of a crash:

      Expected an exit code of 2 but got 0.
      + expected - actual

      -0
      +2
      
      at /home/james-the-hacker/CodeSpace/GitHub-Projects/eslint/tests/bin/eslint.js:30:16
      at processTicksAndRejections (internal/process/task_queues.js:85:5)
      at async Promise.all (index 0)

  5) bin/eslint.js
       handling crashes
         prints the error message pointing to line of code:

      Expected an exit code of 2 but got 0.
      + expected - actual

      -0
      +2
```

Reply:
This is most likely because the tests are running synchronously and so aren’t waiting for `cli.execute()` to finish before asserting the exit code value.